converters/fromZigbee.js

```diff
@@ -307,15 +308,16 @@ const tuyaThermostat = (model, msg, publish, options, meta) => {
         return {eco_temperature: dataAsDecNumber};
     case 621: // 0x6d02 valve position
         return {position: dataAsDecNumber};
-    case 626: // 0x7202 preset temp ?
-        return {preset_temperature: dataAsDecNumber};
-    case 629: // 0x7502 preset ?
-        return {preset: dataAsDecNumber};
-    case 1028: // 0x0404 Mode changed
-        if (utils.getMetaValue(msg.endpoint, model, 'tuyaThermostatSystemMode').hasOwnProperty(dataAsDecNumber)) {
-            return {system_mode: utils.getMetaValue(msg.endpoint, model, 'tuyaThermostatSystemMode')[dataAsDecNumber]};
+    case 626: // 0x7202 away preset temperature
+        return {away_preset_temperature: dataAsDecNumber};
+    case 629: // 0x7502 away preset number of days
+        return {away_preset_days: dataAsDecNumber};
+    case 1028: // 0x0404 Preset changed
+        if (utils.getMetaValue(msg.endpoint, model, 'tuyaThermostatPreset').hasOwnProperty(dataAsDecNumber)) {
```

Comment:
this changes breaks e.g. https://github.com/Koenkk/zigbee-herdsman-converters/blob/eafa4a8dfc9298541827363b06c893297ccce9e2/devices.js#L1529

Reply:
I've forgot to check SYstemMode dictionary. Fixed